compiler/src/iree/compiler/Codegen/Common/GenericVectorization.cpp

```diff
@@ -83,13 +86,81 @@ inferVectorSizesFromIR(linalg::LinalgOp linalgOp) {
   return vectorSizes;
 }
 
-// Return the vector sizes from the local lowering config or try to infer them
+static std::optional<SmallVector<int64_t>>
+inferVectorSizesFromIR(tensor::PackOp op) {
+  LLVM_DEBUG(VEC_DBGS() << "Inferring vector sizes for:\n" << op << "\n");
+
+  if (llvm::any_of(op.getInnerTiles(), [](OpFoldResult v) {
+        return !getConstantIntValue(v).has_value();
+      })) {
+    LLVM_DEBUG(VEC_DBGS() << "skip, because inner_tiles are not all constant");
+    return std::nullopt;
+  }
+
+  SmallVector<int64_t> vectorSizes;
+  auto inferred = inferVectorSizesFromIR(op.getSource());
+  if (!inferred) {
+    return std::nullopt;
+  }
+  vectorSizes = inferred.value();
+
+  for (auto [dimPos, tileSize] :
+       llvm::zip_equal(op.getInnerDimsPos(), op.getStaticInnerTiles())) {
+    if (vectorSizes[dimPos] % tileSize != 0) {
+      return std::nullopt;
+    }
+    vectorSizes[dimPos] /= tileSize;
+  }
+  auto outerDimsPerm = op.getOuterDimsPerm();
+  if (!outerDimsPerm.empty()) {
+    applyPermutationToVector(vectorSizes, outerDimsPerm);
+  }
+
+  LLVM_DEBUG({
+    VEC_DBGS() << "After adjustment with inner tiles and "
+                  "outer_dims_perm:\n";
+    for (auto [idx, val] : llvm::enumerate(vectorSizes)) {
+      llvm::dbgs() << "Dim #" << idx << ": " << val << "\n";
+    }
+  });
+
+  return vectorSizes;
+}
+
+static std::optional<SmallVector<int64_t>> inferVectorSizesFromIR(Value val) {
+  std::optional<SmallVector<int64_t>> result;
+  TypeSwitch<Operation *, void>(val.getDefiningOp())
+      .Case<linalg::LinalgOp, tensor::PackOp>(
+          [&](auto op) { result = inferVectorSizesFromIR(op); })
+      .Case<tensor::ExtractSliceOp>([&](tensor::ExtractSliceOp op) {
+        LLVM_DEBUG(VEC_DBGS() << "Inferring vector sizes for:\n" << op << "\n");
+        SmallVector<int64_t> vectorSizes;
+        int64_t destRank = op.getResult().getType().getRank();
+        for (int dim = 0; dim < destRank; ++dim) {
+          LLVM_DEBUG(VEC_DBGS() << "Dim #" << dim << ": ");
+          FailureOr<int64_t> maybeDimBound =
+              ValueBoundsConstraintSet::computeConstantBound(
+                  presburger::BoundType::UB, op, dim,
+                  /*stopCondition=*/nullptr, /*closedUB=*/true);
+          if (failed(maybeDimBound)) {
+            LLVM_DEBUG(llvm::dbgs() << "failed\n");
```

Comment:
```suggestion
            LLVM_DEBUG(VEC_DBGS() << "failed\n");
```

Reply:
`VEC_DBGS` inserts one more `[DEBUG_TYPE]`, while I want the `Dim # XXXXX` sentence.